src/compiler/checker.ts

```diff
@@ -39520,7 +39520,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             error(node, Diagnostics.const_enums_can_only_be_used_in_property_or_index_access_expressions_or_the_right_hand_side_of_an_import_declaration_or_export_assignment_or_type_query);
         }
 
-        if (getIsolatedModules(compilerOptions)) {
+        // Do not check for --verbatimModuleSyntax here because imports of ambient const enums

```

Comment:
I don't understand this comment; isn't `getIsolatedModules` just `compilerOptions.isolatedModules || compilerOptions.verbatimModuleSyntax`?

Reply:
the `verbatimModuleSyntax` condition is `&&`'d with “resolves to a global”; dprint would not let me group them together

Reply:
```
compilerOptions.isolatedModules
 || compilerOptions.verbatimModuleSyntax && ok && !(resolveName(node, getFirstIdentifier(node as EntityNameOrEntityNameExp ...
```

Seems to work.